docs/source/serialization.mdx

```diff
@@ -49,255 +60,365 @@ Ready-made configurations include the following models:
+```bash
+Validating ONNX model...
+        -[✓] ONNX model outputs' name match reference model ({'logits'})
+        - Validating ONNX Model output "logits":
+                -[✓] (2, 2) matches (2, 2)
+                -[✓] all values close (atol: 1e-05)
+All good, model saved at: onnx/model.onnx
+```
 
-### Graph conversion
+Notice that in this case, the output names from the fine-tuned model are
+`logits` instead of the `last_hidden_state` we saw with the
+`distilbert-base-uncased` checkpoint earlier. This is expected since the
+fine-tuned model has a sequence classification head.
 
 <Tip>
 
-The approach detailed here is bing deprecated. We recommend you follow the part above for an up to date approach.
+The features that have a `with-past` suffix (e.g. `causal-lm-with-past`)
+correspond to model topologies with precomputed hidden states (key and values
+in the attention blocks) that can be used for fast autoregressive decoding.
 
 </Tip>
 
-Exporting a model is done through the script *convert_graph_to_onnx.py* at the root of the transformers sources. The
-following command shows how easy it is to export a BERT model from the library, simply run:
 
-```bash
-python convert_graph_to_onnx.py --framework <pt, tf> --model bert-base-cased bert-base-cased.onnx
-```
+### Exporting a model for an unsupported architecture
 
-The conversion tool works for both PyTorch and Tensorflow models and ensures:
+If you wish to export a model whose architecture is not natively supported by
+the library, there are three main steps to follow:
 
-- The model and its weights are correctly initialized from the Hugging Face model hub or a local checkpoint.
-- The inputs and outputs are correctly generated to their ONNX counterpart.
-- The generated model can be correctly loaded through onnxruntime.
+1. Implement a custom ONNX configuration.
+2. Export the model to ONNX.
+3. Validate the outputs of the PyTorch and exported models.
 
-<Tip>
+In this section, we'll look at how DistilBERT was implemented to show what's
+involved with each step.
 
-Currently, inputs and outputs are always exported with dynamic sequence axes preventing some optimizations on the
-ONNX Runtime. If you would like to see such support for fixed-length inputs/outputs, please open up an issue on
-transformers.
+#### Implementing a custom ONNX configuration
 
-</Tip>
-
-Also, the conversion tool supports different options which let you tune the behavior of the generated model:
+Let's start with the ONNX configuration object. We provide three abstract
+classes that you should inherit from, depending on the type of model
+architecture you wish to export:
 
-- **Change the target opset version of the generated model.** (More recent opset generally supports more operators and
-  enables faster inference)
+* Encoder-based models inherit from [`OnnxConfig`]
```

Comment:
I'm not entirely sure how cross-referencing works with the new doc - looking at other sections, it seems square brackets are used. Since `OnnxConfig` is not visible in the docs page, perhaps I should not do this here?

Reply:
I've opted to link to the source code for these classes as they're not currently visible on the docs. (Will add that in a separate PR)